samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -174,6 +176,7 @@
   private final MetadataStore metadataStore;
 
   private final SystemAdmins systemAdmins;
+  private final LocalityManager localityManager;
```

Comment:
Here is another thought:
LocalityManager and the ContainerPlacementMetadataStore both are present in this class ClusterBasedJobCoordinator because they both need a hold of metadata store

Will it be cleaner to move this to ContainerProcessManager and then eventually move it to ContainerManager (when all the refactors we have in process are in)

We can just pass metadata store from here and then instantiate ContainerPlacementMetadatStore and LocalityManager in ContainerManager

Reply:
In the first pass, I didn't have a local instance for locality manager but ended up introducing one to manage the lifecycle along with the other components. 

What do you think about doing it as part of the consolidation refactor work? We currently have `ContainerPlacementMetadataStore` in job coordinator. when we move that within CM, we can also move locality manager all in on go.

Reply:
Sure we can take this work as a part of consolidation refactor, please add a todo in code so we do not forget that